frontend/js/src/explore/artist-similarity/artist-search/SearchBox.tsx

```diff
@@ -0,0 +1,181 @@
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import {
+  faMagnifyingGlass,
+  faMinus,
+  faPlus,
+} from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { throttle } from "lodash";
+import SearchDropdown from "./SearchDropdown";
+import artistLookup, { ArtistType } from "./artistLookup";
+
+interface SearchBoxProps {
+  currentsimilarArtistsLimit: number;
+  onSimilarArtistsLimitChange: (similarArtistsLimit: number) => void;
+  onArtistChange: (artist_mbid: string) => void;
+}
+
+function SearchBox({
+  currentsimilarArtistsLimit,
+  onSimilarArtistsLimitChange,
+  onArtistChange,
+}: SearchBoxProps) {
+  // State to store the search results (list of artists)
+  const [searchResults, setSearchResults] = React.useState<Array<ArtistType>>(
+    []
+  );
+  const [searchQuery, setSearchQuery] = React.useState<string>("");
+  // State to toggle the dropdown menu for search
+  const [openDropdown, setOpenDropdown] = useState(false);
+
+  /* Below is the piece of the code with problem.
+    *   My plan: 
+    *       I decided to make the api call outside of this component in artistLookup, a bit different from SearchTrackorMBID.
+    *       Thinking it would decrease the complexity of the code but I guess I added more unintentionally.
+    *   Problem:
+    *       Oftentimes the results presented in the dropdown would lag behind the search query e.g. Bruno Mars even though the api call
+    *       was made correctly. I believe the problem is how React re-renders on state changes but I am very unsure. In this part I use 
+    *       throttle in the artistLookup.
+    * /
+    
+    /*
+    // Lookup the artist based on the query
+    const getArtists = async (): Promise<void> => {
+        if(searchQuery.length && searchQuery.trim() !== ""){
+            let results = await artistLookup(searchQuery);
+            setSearchResults(results ?? []);
+            //console.log(searchResults);
+            // Open the dropdown if any results exist
+            if(searchResults.length) {
+                //console.log(searchResults);
+                setOpenDropdown(true);
+            }
+        }
+        else{
+            setSearchResults([]);
+        }
+    };
+
+    const handleQueryChange = (query: string) => {
+        setSearchQuery(query);
+    }
+
+    useEffect(() => {
+        getArtists();
+    },[searchQuery]);
+    */
+
+  /**
+   * Solution:
+   *  I was able to look up an article for the problem, https://www.developerway.com/posts/debouncing-in-react
+   *  The solution works as expected but I feel like it might be a bit hard to understand what's going on.
+   */
+
+  const getArtists = useCallback(async () => {
+    if (searchQuery.length && searchQuery.trim().length) {
+      const results = await artistLookup(searchQuery);
+      setSearchResults(results);
+      setOpenDropdown(true);
+    }
+  }, [searchQuery]);
+  const getArtistsRef = useRef(getArtists);
```

Comment:
You can remove the useRef and the useEffect below, useCallback will take of updating the function whenever getArtists changes.

Reply:
If I use searchQuery as a dependency for the useCallback in getArtists it will update whenever searchQuery updates and does not work as expected. Therefore, we need to ensure getArtists never gets updated. 
So, a change suggested here, https://www.developerway.com/posts/debouncing-in-react, it says I could send the query directly to the getArtists without keeping any dependencies and it works fine this way. As opposed to, when I use a dependency and causing problems.
I have kept the previous solution as well. Let me know what will be better.

Reply:
I see, I read your code and the article and indeed your way is the correct and the most succint way to do this.